test/Datadog.Trace.Tests/Configuration/ConfigurationSourceTests.cs

```diff
@@ -9,10 +9,28 @@ namespace Datadog.Trace.Tests.Configuration
 {
     public class ConfigurationSourceTests
     {
+        public static IEnumerable<object[]> GetGlobalDefaultTestData()
+        {
+            yield return new object[] { CreateGlobalFunc(s => s.DebugEnabled), false };
+        }
+
+        public static IEnumerable<object[]> GetGlobalTestData()
+        {
+            yield return new object[] { ConfigurationKeys.DebugEnabled, "true", CreateGlobalFunc(s => s.DebugEnabled), true };
```

Comment:
Can you also test that garbage entries like `"Definitely False"` result in `DebugEnabled==false`?